ros/src/computing/planning/motion/packages/waypoint_maker/nodes/waypoint_loader/waypoint_filter.cpp

```diff
@@ -0,0 +1,360 @@
+      const geometry_msgs::Point& p1 = curve_point[1];
+      double theta = fmod(atan2(p1.y - cy, p1.x - cx) - atan2(p0.y - cy, p0.x - cx), 2 * M_PI);
+      if (theta > M_PI)
+        theta -= 2 * M_PI;
+      else if (theta < -M_PI)
+        theta += 2 * M_PI;
+      // interport
+      double t = atan2(p0.y - cy, p0.x - cx);
+      autoware_msgs::waypoint wp;
+      wp.pose.pose.position = lane->waypoints.back().pose.pose.position;
+      wp.change_flag = lane->waypoints.back().change_flag;
+      double dist = radius * fabs(theta);
+      double dz_nextpt = curve_point[1].z - lane->waypoints.back().pose.pose.position.z;
+      const double resample_dz = resample_interval_ * dz_nextpt / dist;
+      for (; dist > resample_interval_; dist -= resample_interval_)
+      {
+        if (lane->waypoints.size() == lane->waypoints.capacity())
+          break;
+        const int sign = (theta > 0.0) ? (1) : (-1);
+        t += sign * resample_interval_ / radius;
+        const double yaw = fmod(t + sign * M_PI / 2.0, 2 * M_PI);
+        wp.pose.pose.position.x = cx + radius * cos(t);
+        wp.pose.pose.position.y = cy + radius * sin(t);
+        wp.pose.pose.position.z += resample_dz;
+        wp.pose.pose.orientation = tf::createQuaternionMsgFromYaw(yaw);
+        lane->waypoints.push_back(wp);
+        curve_radius->push_back(threshold_radius);
+      }
+    }
+    lane->waypoints.back().wpstate = original_lane.waypoints[i].wpstate;
+    lane->waypoints.back().change_flag = original_lane.waypoints[i].change_flag;
+  }
+  lane->waypoints.back().wpstate = original_lane.waypoints.back().wpstate;
+  lane->waypoints.back().change_flag = original_lane.waypoints.back().change_flag;
+}
+
+void WaypointFilter::getCurveAll(const autoware_msgs::lane& lane, std::vector<double>* curve_radius)
+{
+  if (lane.waypoints.empty())
+    return;
+  curve_radius->resize(lane.waypoints.size());
+  curve_radius->at(0) = curve_radius->back() = r_inf_;
+
+  const unsigned int n = (lookup_crv_width_ - 1) / 2;
+  for (unsigned long i = 1; i < lane.waypoints.size() - 1; i++)
+  {
+    std::vector<geometry_msgs::Point> curve_point(3);
+    curve_point[0] = (i < n) ? lane.waypoints[0].pose.pose.position : lane.waypoints[i - n].pose.pose.position;
+    curve_point[1] = lane.waypoints[i].pose.pose.position;
+    curve_point[2] = (i >= lane.waypoints.size() - n) ? lane.waypoints.back().pose.pose.position :
+                                                        lane.waypoints[i + n].pose.pose.position;
+    const std::vector<double> curve_param = getCurveOnce(curve_point);
+    // if going straight
+    if (curve_param.empty())
+    {
+      curve_radius->at(i) = r_inf_;
+    }
+    // else if turnning curve
+    else
+    {
+      const double& radius = curve_param[2];
+      curve_radius->at(i) = (radius > r_inf_) ? r_inf_ : radius;
+    }
+  }
+}
+
+const std::vector<double> WaypointFilter::calcVelParamFromVmax(const double vmax) const
```

Comment:
use std::pair and global value, `velocity_max_`

Reply:
I fixed it to use velocity_max.
However, because we made the calculation expressions easy to understand, unnecessary calculations were reduced, so we no longer need to use std::pair.

Reply:
Good